### PR TITLE
Reuse TPL boolean task results

### PR DESCRIPTION
**Pull request by drewnoakes**

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs [hunk 1]
```diff
@@ -10,13 +10,13 @@ namespace Microsoft.VisualStudio.Threading
     internal static class TaskResult
```
Comment (sharwell) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs]: 💭 Why not just remove this class?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs [hunk 2]
```diff
@@ -10,13 +10,13 @@ namespace Microsoft.VisualStudio.Threading
...
     {
         /// <summary>
-        ///     Represents a Task that's completed successfully with the result of <see langword="false"/>.
+        ///     Represents a <see cref="Task{TResult}"/> that's completed successfully with the result of <see langword="false"/>.
         /// </summary>
-        public static readonly Task<bool> False = Task.FromResult(false);
+        public static readonly Task<bool> False = TplExtensions.FalseTask;
```
Comment (sharwell) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs]:
```suggestion
        public static Task<bool> False => TplExtensions.FalseTask;
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs [hunk 3]
```diff
@@ -10,13 +10,13 @@ namespace Microsoft.VisualStudio.Threading
...
 
         /// <summary>
-        ///     Represents a Task that's completed successfully with the result of <see langword="true"/>.
+        ///     Represents a <see cref="Task{TResult}"/> that's completed successfully with the result of <see langword="true"/>.
         /// </summary>
-        public static readonly Task<bool> True = Task.FromResult(true);
+        public static readonly Task<bool> True = TplExtensions.TrueTask;
```
Comment (sharwell) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs]:
```suggestion
        public static Task<bool> True => TplExtensions.TrueTask;
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs]: It crossed my mind but `TplExtensions` is a mouthful.

Comment (sharwell) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskResult.cs]: 🤪

**Pull request merged**